### PR TITLE
Gamelab animation category previews

### PR DESCRIPTION
**Pull request by caleybrock**
- Create categories when selecting animations to make it easier to find and browse animations
- Made static preview assets and a constant list of categories
- Images uploaded to S3 use the existing alias system to tag animations with a category like `category_animals`
- Breadcrumbs are used to navigate back to the landing screen
- Search within a category gets you only animations in that category
- All category shows everything
- I uploaded generic items and animals with the correct tags in the last PR
- Next PR: I'll add pagination and a default screen incase your search results come up empty

screen shot 2016-10-14 at 10 36 34 am
screen shot 2016-10-14 at 10 36 41 am

### CODE REVIEW

File: apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx [hunk 1]
```diff
@@ -87,10 +135,11 @@ WarningLabel.propTypes = {
  * Given a search query, generate a results list of animationProps objects that
  * can be displayed and used to add an animation to the project.
  * @param {string} searchQuery - text entered by the user to find an animation
+ * @param {string} categoryQuery - name of category user selected to filter animations
  * @return {Array.<SerializedAnimationProps>} - Limited list of animations
  *         from the library that match the search query.
  */
-function searchAnimations(searchQuery) {
+function searchAnimations(searchQuery, categoryQuery) {
```
Comment (caleybrock) [inline apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx:144]: I think there will be continuous improvement on this function, particularly to help make animations easier to find, but I'm definitely open to ideas about making this fasters. Right now I'm just getting search results and category results and then taking the intersection.

File: apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx [hunk 2]
```diff
@@ -3,14 +3,24 @@ import React from 'react';
...
 import ScrollableList from '../AnimationTab/ScrollableList.jsx';
 import styles from './styles';
 import AnimationPickerListItem from './AnimationPickerListItem.jsx';
 import AnimationPickerSearchBar from './AnimationPickerSearchBar.jsx';
 
-const MAX_SEARCH_RESULTS = 18;
+const MAX_SEARCH_RESULTS = 40;
+const allAnimationsStyle = {
+  color: color.purple,
+  fontFamily: "'Gotham 7r', sans-serif, sans-serif",
```
Comment (Bjvanminnen) [inline apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx]: sans-serif twice intentionally?

File: apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx [hunk 3]
```diff
@@ -3,14 +3,24 @@ import React from 'react';
...
+  cursor: "pointer"
+};
+const breadCrumbsStyle = {
+  margin: "8px 0",
+  fontSize: 14
+};
```
Comment (Bjvanminnen) [inline apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx:25]: Not sure if you guys take this approach in GL, but in much of our code, we've been putting all of our styles in a single obj, i.e.

```
const styles = {
  allAnimations: {
    color: color.purple,
    ...
  },
  breadCrumbs: {
    margin: "8x 0"
    fontSize: 14
  }
}
```

File: apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx [hunk 4]
```diff
@@ -23,16 +33,47 @@ const AnimationPickerBody = React.createClass({
...
     };
   },
 
   onSearchQueryChange(value) {
     this.setState({searchQuery: value});
   },
 
+  onCategoryChange(category) {
+    this.setState({categoryQuery: category});
+  },
```
Comment (Bjvanminnen) [inline apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx:49]: One thing you could do with these is name your param the same as your state field, then take advantage of an ES6 feature, where `{ foo }` is the same thing as `{ foo: foo }`. Up do you which approach you prefer.

```
onCategoryChange(categoryQuery) {
  this.setState({categoryQuery});
}
```

File: apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx [hunk 5]
```diff
@@ -23,16 +33,47 @@ const AnimationPickerBody = React.createClass({
...
+    this.setState({categoryQuery: '', searchQuery: ''});
+  },
+
+  animationCategoriesRendering() {
+    return Object.keys(AnimationCategories).map(category =>
+      <AnimationPickerListItem
+        key={category}
+        label={AnimationCategories[category]}
+        category={category}
+        onClick={() => this.onCategoryChange(category)}
```
Comment (Bjvanminnen) [inline apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx]: Might be micro-optimizing, but every time you render this, you'll create a new function object. Another option might be to have `onCategoryChange` extract the category from `event.target`, allowing us to just do `onClick={this.onCategoryChange}`

File: apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx [hunk 6]
```diff
@@ -23,16 +33,47 @@ const AnimationPickerBody = React.createClass({
...
+      />
+    );
+  },
+
+  animationItemsRendering() {
+    var pageOfResults = searchAnimations(this.state.searchQuery, this.state.categoryQuery);
```
Comment (Bjvanminnen) [inline apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx]: s/var/const

File: apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx [hunk 7]
```diff
@@ -47,26 +88,33 @@ const AnimationPickerBody = React.createClass({
           value={this.state.searchQuery}
           onChange={this.onSearchQueryChange}
         />
+        {this.state.categoryQuery !== '' &&
+          <div style={breadCrumbsStyle}>
+            <span onClick={this.onClearCategories} style={allAnimationsStyle}>{"All categories > "}</span>
```
Comment (Bjvanminnen) [inline apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx]: do we need to worry about i18n?

File: apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx [hunk 8]
```diff
@@ -100,12 +149,25 @@ function searchAnimations(searchQuery) {
...
 
   // Generate the set of all results associated with all matched aliases
-  const resultSet = Object.keys(animationLibrary.aliases)
+  let resultSet = Object.keys(animationLibrary.aliases)
       .filter(alias => searchRegExp.test(alias))
       .reduce((resultSet, nextAlias) => {
         return resultSet.union(animationLibrary.aliases[nextAlias]);
       }, Immutable.Set());
 
+  if (categoryQuery !== '' && categoryQuery !== 'category_all') {
```
Comment (Bjvanminnen) [inline apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx]: Should `category_all` be a constant somewhere?

File: apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx (hunk 7 above)
Comment (caleybrock) [reply apps/src/gamelab/AnimationPicker/AnimationPickerBody.jsx]: Right now we're just using english strings.

**Pull request merged**